nilber-calculator/src/Components/Calculator.js

```diff
@@ -0,0 +1,96 @@
+import React, {useState} from 'react';
+import Button from './Button';
+import Error from './Error';
+import '../CSS/Calculator.css'
+import { create, all } from "mathjs";
+import classNames from 'classnames'
+
+const math = create(all);
+
+const Calculator = () => {
+
+    const [computation, setComputation] = useState("")
+    const [result, setResult] = useState("0")
+    const [error, setError] = useState(false)
+
+    const handleClear = () => {
+        setComputation("")
+        setResult("0")
+        setError(false)
+    }
+
+    const handleBackspace = () => {
+        setComputation(computation.slice(0,computation.length-1))
+    }
+
+    const handleCalculate = () => {
+        try {
+            let solution = math.evaluate(computation)
+            setResult(solution)
+            setComputation("")
+        } catch (error) {
+            setResult("")
+            setComputation("")
+            setError(true)
+        }
+    }
+    
+    const handleSign = () => {
+        computation.charAt(0) === "-" ? setComputation(computation.slice(1)) : setComputation("-" + computation)
+    }
+    
+    const handlePercentage = () => {
+        setComputation(computation / 100)
+    }
+
+    const handleExpression = (event) => {
+        setComputation(computation + event.target.value)
+    }
+
+    let calculatorKeys = [
+        {name: "obliterate", value: "Clear", handleClick: handleClear},
+        {name: "backspace", value:"<-", handleClick: handleBackspace},
+        {name: "equal", value:"=", handleClick: handleCalculate},
+        {name: "sign", value:"-+", handleClick: handleSign},
+        {name: "percentage", value: "%", handleClick: handlePercentage},
+        {name: "plus", value: "+", handleClick: handleExpression},
+        {name: "subtract", value: "-", handleClick: handleExpression},
+        {name: "multiply", value: "*", handleClick: handleExpression},
+        {name: "divide", value: "/", handleClick: handleExpression},
+        {name: "decimal", value: ".", handleClick: handleExpression},
```

Comment:
Maybe the decimal point deserved a separate function since you can't just keep adding that decimal point to the screen. It allows to add several of them. You might want to check if the string already includes a decimal and not allow adding the second one.

Reply:
Gotcha, I don't believe the string already adds a decimal. I was leaving it up to the try/catch to throw errors for any invalid syntax and allow the user to clear the error.